app/lib/bk/compat/parsers/bitrise/github_release.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module BK
+  module Compat
+    module BitriseSteps
+      # Handles the translation of github-release
+      class Translator
+        def translate_github_release(inputs)
+          raise 'Invalid github-release step configuration!' unless validate_required_params(inputs)
+
+          release_config = extract_release_config(inputs)
+          owner, repo_name = parse_repo(release_config[:repository_url])
+
+          commands = build_commands(release_config, owner, repo_name)
+          BK::Compat::CommandStep.new(commands: commands)
+        end
+
+        private
+
+        def validate_required_params(inputs)
+          %w[api_token username repository_url tag commit name].all? { |key| inputs.include?(key) }
+        end
+
+        def extract_release_config(inputs)
+          {
+            api_token: inputs['api_token'],
+            commit: inputs['commit'],
+            name: inputs['name'],
+            repository_url: inputs['repository_url'],
+            tag: inputs['tag'],
+            body: inputs['body'] || '',
+            draft: inputs['draft'] || 'yes',
+            pre_release: inputs['pre_release'] || 'no',
+            files_to_upload: inputs['files_to_upload'],
+            api_base_url: inputs['api_base_url'] || 'https://api.github.com',
+            upload_base_url: inputs['upload_base_url'] || 'https://uploads.github.com',
+            generate_release_notes: inputs['generate_release_notes'] || 'no'
+          }
```

Comment:
A cleaner way to provide the defaults would be so that you don't have to add all the keys that are mandatory:

```suggestion
          defaults = {
            'body' => '',
            'draft' => 'yes',
            'pre_release' => 'no',
            'api_base_url' => 'https://api.github.com',
            'upload_base_url' => 'https://uploads.github.com',
            'generate_release_notes' => 'no'
          }
          
          defaults.merge(inputs)
```